cypress/e2e/signIn.cy.js

```diff
@@ -10,29 +9,37 @@ const homePage = new HomePageObject();
 describe('Sign In page', () => {
   let user;
 
-  before(() => {
+  beforeEach(() => {
     cy.task('db:clear');
     cy.task('generateUser').then(generateUser => {
       user = generateUser;
     });
   });
 
   it('should provide an ability to log in with existing credentials', () => {
-    signInPage.visit();
     cy.register(user.email, user.username, user.password);
```

Comment:
It seems you are registering the user in each test case. If registration is not part of what you are testing, consider moving it to a 'before' or 'beforeEach' hook to avoid repeating code.